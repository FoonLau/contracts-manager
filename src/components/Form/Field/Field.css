.field {
  margin-bottom: 15px;
}

.field-input {
  width: 100%;
  border: 1px solid #d4d4d4;
  border-radius: 3px;
  height: 30px;
}

.field-error {
  color: #ab7277;
}
