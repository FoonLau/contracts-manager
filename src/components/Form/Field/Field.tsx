import * as React from 'react';
import './Field.css';

export interface Props {
  name: string;
  label: string;
  type: string;
  value: string;
}

const Field: React.StatelessComponent<Props> = ({
  label,
  type,
  name,
  value
}) => {
  return (
    <div className="field">
      <label className="field-label">{label}</label>
      <div>
        <input 
          name={name}
          className="field-input"
          placeholder={label} 
          type={type} 
          value={value} />
      </div>
    </div>
  );
}

export default Field;
