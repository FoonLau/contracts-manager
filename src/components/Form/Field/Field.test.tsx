import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Field from './Field';

it('should render correctly', () => {
  const tree = renderer.create(<Field name="name" label="name" type="text" value="John" />);

  expect(tree).toMatchSnapshot();
});
