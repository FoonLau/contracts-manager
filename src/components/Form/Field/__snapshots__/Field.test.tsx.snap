// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="field"
>
  <label
    className="field-label"
  >
    name
  </label>
  <div>
    <input
      className="field-input"
      name="name"
      placeholder="name"
      type="text"
      value="John"
    />
  </div>
</div>
`;
